plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,286 @@
+     */
+    @Test
+    public void exportImportRoundTrip() throws IOException {
+        String resourcePath = configResource();
+        String resourceContent = getResourceContent(resourcePath);
+
+        assertNotNull(resourcePath);
+        assertNotNull(resourceContent);
+
+        r.then(step -> {
+            // Configure and validate
+            configureWithResource(resourcePath);
+            assertConfiguredAsExpected(r, resourceContent);
+
+            // Get the full configuration
+//            String jenkinsConf = getJenkinsConfViaWebUI();
+
+            // Here we check that the exported configuration has the same information as the YAML used.
+            // hack: But when JCasC is running under test (not normally) the ciphered secrets in the YAML are ciphered
+            // again, so the exported configuration won't be the same never. When running normally it works. When the
+            // bug filed is solved we can uncomment that check. In other way, all the plugins with plain secrets in
+            // their YAMLs will fail in this test. Example: Credentials.
+            //TODO: uncomment when https://issues.jenkins-ci.org/browse/JENKINS-57863 is solved
+            //assertJenkinsHasConfiguredPath(jenkinsConf, resourcePath);
+
+            //hack: the full Jenkins config fails due to defaultProperties of maven, we use the config of the plugin,
+            //but first, we guarantee that the exported configuration is the same as the configured.
+            //TODO: remove when https://issues.jenkins-ci.org/browse/JENKINS-57122 is solved
+            //@Issue("JENKINS-57122")
+            String jenkinsConf = getResourceContent(resourcePath);
+
+            // Check if the exported configuration is valid
+            assertConfigViaWebUI(jenkinsConf);
+
+            // Apply full configuration. Maybe not needed, we already have it configured and checked it is valid.
+            applyConfigViaWebUI(jenkinsConf);
+            assertConfiguredAsExpected(r, resourceContent);
+
+            // Configure Jenkins default JCasC file with the config file. It's already established, we check if applied
+            // looking at the logs.
+            putConfigInHome(jenkinsConf);
+
+            // Start recording the logs just before restarting, to avoid capture the previous startup. We're look there
+            // if the "magic token" is there
+            logging.record("io.jenkins.plugins.casc", Level.INFO).capture(5);
+        });
+
+        r.then(step -> {
+            // Verify the log shows it's configured
+            assertLogAsExpected(stringInLogExpected());
+
+            // Verify the configuration set at home/jenkins.yaml is loaded
+            assertConfiguredAsExpected(r, resourceContent);
+        });
+    }
+
+//    private void assertJenkinsHasConfiguredPath(String jenkinsConfig, String resourcePath) throws IOException, URISyntaxException {
```

Comment:
If you comment out a big chunk of code, please put explanation, or a link to the discussion in comments before it.
Bonus points for TODO

Reply:
done